logstash-core/lib/logstash/agent_plugin_manager.rb

```diff
@@ -0,0 +1,65 @@
+# encoding: utf-8
+require "logstash/agent"
+
+module LogStash module AgentPluginManager
+  DEFAULT_AGENT_NAME = :default
+  class DuplicatePluginNameError < Error; end
+
+  REGISTRY = {}
+
+  # Reset plugin registry to just the default plugin
+  def self.reset!
+    REGISTRY.clear
+    REGISTRY[DEFAULT_AGENT_NAME] = LogStash::Agent
+  end
+
+  reset!
```

Comment:
I strongly think we should not use this, all calls should go into methods/classes. We can not have this kind of uncontrolled changes to status. Please avoid using it.